tools/deploy.py

```diff
@@ -53,17 +49,36 @@ def deploy_pew_pew_control_module(containerPath):
 def deploy_py3status_module():
     configHomePath = Path(os.getenv("XDG_CONFIG_HOME", "~")).expanduser()
     candidates = [
-        configHomePath / ".config/py3status/modules",
-        configHomePath / ".config/i3status/py3status",
-        configHomePath / ".config/i3/py3status",
-        configHomePath / ".i3/py3status",
+        configHomePath / ".config" / "py3status" / "modules",
+        configHomePath / ".config" / "i3status" / "py3status",
+        configHomePath / ".config" / "i3" / "py3status",
+        configHomePath / ".i3" / "py3status",
     ]
     path = first_existing(candidates)
     if path is not None:
         log.info(f"deploy py3status module {PY3STATUS_MODULE_PATH} to {path}")
         shutil.copy(PY3STATUS_MODULE_PATH, path)
     else:
-        raise Exception("Could not find a home for pewpew")
+        raise Exception("Could not find a home for {}".format(MODULE_NAME))
+
+
+def warn_if_user_not_in_expected_groups():
+    user = getpass.getuser()
+    groups = [g.gr_name for g in grp.getgrall() if user in g.gr_mem]
+    expected_groups = ['input', 'dialout']
+    for exp_group in expected_groups:
+        if exp_group not in groups:
+            print("Warning: user '{}' is not in expected group: '{}'".format(user, exp_group))
+
+
+def main():
```

Comment:
I know this is an unusual view but I really like the main being at the beginning of the file as it is like the high level abstract of what is happening, which one usually reads first. I never understood why it is a common practice to place the main() at the bottom of the file.

Reply:
I've seen others preferring top-line, and in fact when I read such files (not write) I prefer this practice, it's just more readable!